packages/aws-cdk/lib/commands/flag-operations.ts

```diff
@@ -150,12 +143,30 @@ async function prototypeChanges(
     outdir: path.join(process.cwd(), 'original'),
   });
 
-  const cx = await toolkit.synth(source);
-  const assembly = cx.cloudAssembly;
-
   const updateObj: Record<string, boolean> = {};
-  updateObj[flagName] = boolValue;
+  if (typeof (flagNames) == 'string') {
+    if (baseContextValues[flagNames] == value) {
+      await ioHelper.defaults.error('Flag is already set to the specified value. No changes needed.');
```

Comment:
I don't think we need to give this feedback to the user. The expectation is that the operation is idempotent: applying it twice is not an error. If we don't need to do anything, just return quietly.